plugins/polio/js/src/components/LQAS-IM/RefreshLqasData.tsx

```diff
@@ -0,0 +1,149 @@
+import React, {
+    FunctionComponent,
+    useCallback,
+    useEffect,
+    useState,
+} from 'react';
+import { useSafeIntl, LoadingSpinner } from 'bluesquare-components';
+import moment from 'moment';
+import { Box, Button } from '@material-ui/core';
+import RefreshIcon from '@material-ui/icons/Refresh';
+import { useQueryClient } from 'react-query';
+import {
+    Task,
+    TaskApiResponse,
+} from '../../../../../../hat/assets/js/apps/Iaso/domains/tasks/types';
+import { useGetLatestLQASIMUpdate } from '../../hooks/useGetLatestLQASIMUpdate';
+import {
+    useCreateTask,
+    useTaskMonitor,
+} from '../../../../../../hat/assets/js/apps/Iaso/hooks/taskMonitor';
+import MESSAGES from '../../constants/messages';
+
+type Props = {
+    countryId?: number;
+    category: string;
+};
+
+const LQAS_TASK_ENDPOINT = '/api/polio/tasks/refreshlqas/';
+
+const useLastUpdate = (
+    lastUpdate: Task<any>,
+): { message: string; updateStatus: string } => {
+    const { formatMessage } = useSafeIntl();
+    const result = { message: '', updateStatus: '' };
+    if (lastUpdate?.status === 'RUNNING') {
+        result.message = formatMessage(MESSAGES.ongoing);
+    }
+    if (lastUpdate?.status === 'SUCCESS' && Boolean(lastUpdate?.ended_at)) {
+        result.message = `${moment
+            .unix(lastUpdate.ended_at as number)
+            .format('LTS')}`;
+    }
+    if (lastUpdate?.status) {
+        const updateStatus = formatMessage(MESSAGES[lastUpdate.status])
+            ? formatMessage(MESSAGES[lastUpdate.status])
+            : '';
```

Comment:
this will raise a warning anyway, you should check the presence of the entry in the object MESSAGE
```suggestion
        const updateStatus = MESSAGES[lastUpdate.status]
            ? formatMessage(MESSAGES[lastUpdate.status])
            : '';
```